packages/graphql/src/schema-model/generate-model.ts

```diff
@@ -16,79 +16,117 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-import type {
-    DirectiveNode,
-    DocumentNode,
-    FieldDefinitionNode,
-    ObjectTypeDefinitionNode,
-    UnionTypeDefinitionNode,
-    ValueNode,
-} from "graphql";
-import { Kind } from "graphql";
+import type { DirectiveNode, DocumentNode, FieldDefinitionNode, ObjectTypeDefinitionNode } from "graphql";
+import { Neo4jGraphQLSchemaValidationError } from "../classes";
 import { SCALAR_TYPES } from "../constants";
 import { getDefinitionNodes } from "../schema/get-definition-nodes";
 import getFieldTypeMeta from "../schema/get-field-type-meta";
 import { filterTruthy } from "../utils/utils";
 import type { Annotation } from "./annotation/Annotation";
-import { CypherAnnotation } from "./annotation/CypherAnnotation";
 import { Attribute } from "./attribute/Attribute";
 import { CompositeEntity } from "./entity/CompositeEntity";
 import { ConcreteEntity } from "./entity/ConcreteEntity";
 import { Neo4jGraphQLSchemaModel } from "./Neo4jGraphQLSchemaModel";
+import { parseAuthorizationAnnotation } from "./parser/authorization-annotation";
+import { parseCypherAnnotation } from "./parser/cypher-annotation";
+import { parseArguments } from "./parser/utils";
 
 export function generateModel(document: DocumentNode): Neo4jGraphQLSchemaModel {
     const definitionNodes = getDefinitionNodes(document);
+
+    // init interface to typennames map
+    const interfaceToImplementingTypeNamesMap = definitionNodes.interfaceTypes.reduce((acc, entity) => {
+        const interfaceTypeName = entity.name.value;
+        acc.set(interfaceTypeName, []);
+        return acc;
+    }, new Map<string, string[]>());
+
+    // hydrate interface to typennames map
+    definitionNodes.objectTypes.forEach((el) => {
+        if (!el.interfaces) {
+            return;
+        }
+        const objectTypeName = el.name.value;
+        el.interfaces?.forEach((i) => {
+            const interfaceTypeName = i.name.value;
+            const before = interfaceToImplementingTypeNamesMap.get(interfaceTypeName);
```

Comment:
`before` doesn't seem to be a descriptive name for this variable - I can't tell what it actually means.